src/rev_ai/speechrec.py

```diff
@@ -9,67 +9,85 @@
 import requests
 import logging
 import json
+from models import *
 
 LOG = logging.getLogger(__name__)
 
+class RevSpeechAPIClient:
+    """Client which implements rev.ai API"""
+
+    VERSION = "v1"
+    """Default version of rev.ai"""
 
-class RevSpeechAPI:
-    VERSION = "v1beta"
     BASE_URL = "https://api.rev.ai/revspeech/{v}/".format(v=VERSION)
+    """Default address of the API"""
+
+    def __init__(
+        self,
+        api_key
+    ):
+        """
+        Constructor
 
-    def __init__(self, api_key, use_stage=False):
+        :param api_key: api key which authorizes all requests and links them to your account.
+            Generated on the settings page of your account dashboard on rev.ai
+        """
         if not api_key:
             raise ValueError("API Key cannot be empty.")
         self.s = requests.Session()
         self.s.headers.update({
             'Authorization': 'Bearer {api_key}'.format(api_key=api_key)
         })
-        if use_stage:
-            self.BASE_URL = ("https://api-stage.rev.ai/revspeech/{v}/"
-                             .format(v=self.VERSION))
 
 
-    def submit_job_url(self, media_url, metadata="", callback_url=None):
+    def submit_job_url(
+        self, 
+        media_url, 
+        options
+    ):
         """Submit a web URL for transcription.
         The audio data is downloaded from the URL.
 
         :param media_url: web location of the media file
-        :param metadata: info to associate with the transcription job
-        :param callback_url: callback url to invoke on job completion as a webhook
+        :param options: JobSubmitOptions object for the job
         :returns: raw response data
         """
         url_jobs = urljoin(self.BASE_URL, "jobs")
         payload = {
             'media_url': media_url,
-            'metadata': metadata
+            'metadata': options.metadata
         }
-        if callback_url:
-            payload['callback_url'] = callback_url
+        if options.callback_url:
+            payload['callback_url'] = options.callback_url
         response = self.s.post(url_jobs, json=payload)
 
-        return response.json()
-
-    def submit_job_local_file(self, filename,
-                              media_type="audio", content_type=None, metadata="", callback_url=None):
+        if (response.status_code == 200):
+            return Job.from_json(response.json())
+        elif (response.status_code == 400):
```

Comment:
what about 401 errors. Should we catch them on creation of the client? or on the method?

Reply:
I guess then what about 5xx responses? I'd imagine `ApiError.from_json` isn't going to work

Reply:
As part of a conversation with Josh it became clear that the standard in python is to write the method without errors in mind and let them pop up as actual errors instead of seemingly valid responses. We can talk about this more if you'd like.